hifireg/registration/views/registration.py

```diff
@@ -57,18 +58,33 @@ def register_comp_code(request):
     return render(request, 'registration/register_comp_code.html', {'form': form})
 
 
-@must_have_registration
-def register_policy(request):
-    if request.method == 'POST':
+# @must_have_registration
+# def register_policy(request):
+#     if request.method == 'POST':
+#         if 'previous' in request.POST:
+#             return redirect('register_comp_code')
+#         form = RegPolicyForm(request.POST, instance=Registration.objects.get(user=request.user))
+#         if form.is_valid():
+#             form.save()
+#             return redirect('register_ticket_selection')
+#     else:
+#         form = RegPolicyForm(instance=Registration.objects.get(user=request.user))
+#     return render(request, 'registration/register_policy.html', {'form': form})
+
+@method_decorator(must_have_registration, name='dispatch')
+class RegisterPolicyView(View):
```

Comment:
This is cool and I'm glad you found this strategy, however, along with the CBV strategy, we want to use Mixins instead of decorators because they give us some extra power. The view mixins are all in "mixins.py" that would mean you'd do this instead:
```suggestion
class RegisterPolicyView(RegistrationRequiredMixin, View):
```